rpcs3/Emu/RSX/Overlays/overlay_osk.cpp

```diff
@@ -254,16 +275,25 @@ namespace rsx
 			auto on_accept = [&]()
 			{
 				const u32 current_index = (selected_y * num_columns) + selected_x;
-				const u32 output_count = ::size32(m_grid[current_index].outputs);
+				const auto current_cell = m_grid[current_index];
+
+				std::vector<std::u32string> outputs;
+
+				if (m_selected_mode < layer_mode::mode_count && m_selected_mode < current_cell.outputs.size())
+				{
+					outputs = current_cell.outputs[m_selected_mode];
+				}
```

Comment:
This block and a similar one elsewhere are copying an array of strings every time a button is pressed which seems pointless and wasteful. You could literally flag a bool here to true and use that for the condition below.

Reply:
Fixed by using output_count